src/erlcloud_aws.erl

```diff
@@ -331,6 +331,20 @@ encode_params(Params, Headers) ->
     _ContentType -> {Params, LowerCaseHeaders}
   end.
 
+raw_xml_response(Body) ->
+    case format_xml_response(Body) of
+        {ok, XML} -> XML;
+        Error -> Error
+    end.
+
+format_xml_response(Body) ->
+    try element(1, xmerl_scan:string(binary_to_list(Body))) of
+        XML -> {ok, XML}
+    catch
+        exit:{fatal,{expected_element_start_tag, _, _, _}} ->
+            {aws_error, {invalid_xml_response_document, Body}}
```

Comment:
I don't think `{aws_error, _}` is a valid form here; it should be `{error, {aws_error, {invalid_xml_response_document, Body}}}` IMO. when raising an error, it should be raised as `{aws_error, Reason}`, but this isn't consistent with the other library forms and might mess some consumers up.

Reply:
Error handling in `erlcloud_sns` matches `{error, Reason}` and raises that as `{sns_error, Reason}` . The end result of an invalid xml response document in `aws_request` then becomes ` {sns_error,{aws_error,{invalid_xml_response_document,<<>>}}}` . I believe this is the response that conforms with current error handling in `erlcloud_sns` as everything that fails out of that module raises `{sns_error, Reason}` .